subsys/bluetooth/mesh/sensor_cli.c

```diff
@@ -833,18 +890,24 @@ int bt_mesh_sensor_cli_get(
 	bt_mesh_model_msg_init(&msg, BT_MESH_SENSOR_OP_GET);
 	net_buf_simple_add_le16(&msg, sensor->id);
 
-	struct sensor_data_rsp rsp_data = { .id = sensor->id, .value = rsp };
+	struct bt_mesh_sensor_data sensor_data = {
+		.type = sensor,
+	};
+	struct sensor_data_list_rsp rsp_data = { .count = 1,
+						 .sensors = &sensor_data };
 
 	err = model_ackd_send(cli->mod, ctx, &msg, rsp ? &cli->ack : NULL,
-			       BT_MESH_SENSOR_OP_STATUS, &rsp_data);
+			      BT_MESH_SENSOR_OP_STATUS, &rsp_data);
 	if (err) {
 		return err;
 	}
 
-	if (rsp && !rsp_data.value) {
+	if (rsp && sensor_data.type->id == BT_MESH_PROP_ID_PROHIBITED) {
```

Comment:
This and the code on line 145 doesn't match, and this function can't return `-ENODEV`, since we never set the ID. I think the status handler should set `rsp->sensors[0].type = NULL` before calling `model_ack_rx()`, and this line should be `if (rsp && !sensor_data.type) {`.